File: binarytreeimplementation.java
public class Main {
  int data;
  Main left, right;
  public Main(int data){
    this.data = data;
  }
  public void insert(int value){
    if(value<=data){
      if(left == null){
        left = new Main(value);
      }
      else{
        left.insert(value);
      }
    }
    else{
      if(right == null){
        right = new Main(value);
      }
      else{
        right.insert(value);
      }
    }
  }
  public boolean contains(int value){
    if(value == data){
      return true;
    }
    else if(value<data){
      if(left == null){
        return false;
      }
      else{
        return left.contains(value);
      }
    }
    else{
      if(right==null){
        return false;
      }
      else{
        System.out.println("yes");
        return right.contains(value);
      }
    }
  }
  public void inOrder(){
    if(left!=null){
      left.inOrder();
    }
    System.out.println(data);
    if(right!=null){
      right.inOrder();
    }
  }
    public void preOrder(){
    if(left!=null){
      left.preOrder();
    }
    System.out.println(data);
    if(right!=null){
      right.preOrder();
    }
  }
    public void postOrder(){
    if(left!=null){
      left.postOrder();
    }
    System.out.println(data);
    if(right!=null){
      right.postOrder();
    }
  }
  public static void main(String[] args) {
    Main n1 = new Main(50);
    for(int i=10;i<100;i=i+10){
      n1.insert(i);
    }
    n1.inOrder();
    // n1.preOrder();
    // n1.postOrder();
  }
}

File: binarytreeimplementation.cpp
/////This isn't complete.. refer to the https://github.com/dhirajdkv/Data-Structures/edit/master/binarytreejava.java
#include <iostream>

class Node
{
int data;
Node* left;
Node* right;
};

int AddNode(int data);
Node* root;

int main() {
  std::cout << "Hello World!\n";
}

int AddNode(int data)
{
  if(root==null)
  {
    root = new Node(data);
  }
  else
  {
     Node temp = root;
      while(temp !=NULL)
      {
        if(temp->data<data)
        {
          temp = temp->left;
        }
        else{
          temp=temp->right;
        }
      }

      n1 = new Node(data);
       if(temp->data<data)
        {
          temp->left=n1;
        }
        else{
          temp->right=n1;
        }


  }
}
